src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1890,3 +1893,24 @@ fn is_primitive(path_str: &str, ns: Namespace) -> Option<(&'static str, Res)> {
 fn primitive_impl(cx: &DocContext<'_>, path_str: &str) -> Option<&'static SmallVec<[DefId; 4]>> {
     Some(PrimitiveType::from_symbol(Symbol::intern(path_str))?.impls(cx.tcx))
 }
+
+fn strip_generic_parameters(path_str: &str) -> String {
+    let mut stripped_path_string = String::new();
+
+    let mut param_depth = 0;
+
+    for c in path_str.chars() {
+        if c == '<' {
+            param_depth += 1;
+        } else if c == '>' {
+            param_depth -= 1;
+        } else if param_depth == 0 {
+            stripped_path_string.push(c)
+        }
+    }
+
+    // Should we throw an error here if `param_depth` != 0? That would mean the link is malformed,
+    // e.g. `Vec<T` or `Vec<T>>` or something like that.
```

Comment:
That seems reasonable. You could add a new `ResolutionError::InvalidGenerics` variant to https://github.com/rust-lang/rust/blob/8a13fc494d4365e57d8f343219f0201458b68591/src/librustdoc/passes/collect_intra_doc_links.rs#L62
You'll also need to update `resolution_failure` to cover the new error: https://github.com/rust-lang/rust/blob/8a13fc494d4365e57d8f343219f0201458b68591/src/librustdoc/passes/collect_intra_doc_links.rs#L1522

Ideally it would update `link_range`, but none of the other errors do, so no need to change that if it's hard.

Reply:
What fields should I put on the error variant?